pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -553,6 +557,10 @@ func (r *repository) rememberLayersOfImage(image *imageapi.Image, cacheName stri
 		for _, layer := range image.DockerImageLayers {
 			r.cachedLayers.RememberDigest(digest.Digest(layer.Name), r.blobrepositorycachettl, cacheName)
 		}
+		// remember reference to manifest config as well for schema 2
+		if image.DockerImageManifestMediaType == schema2.MediaTypeManifest && len(image.DockerImageMetadata.ID) > 0 {
+			r.cachedLayers.RememberDigest(digest.Digest(image.DockerImageMetadata.ID), r.blobrepositorycachettl, cacheName)
```

Comment:
> is image.DockerImageMetadata.ID computed when the registry creates an image, or is it provided by the user?

For schema 2, it is parsed from manifest which is immutable: https://github.com/openshift/origin/blob/e22f4b2a0d0caf0f596dc8227c6d02d571e66871/pkg/image/api/helper.go#L543